trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/generate_vortex_distribution.py

```diff
@@ -0,0 +1,1413 @@
+    fhs_xb1 = np.zeros(n_cw*n_sw)
+    fhs_yb1 = np.zeros(n_cw*n_sw)
+    fhs_zb1 = np.zeros(n_cw*n_sw)
+    fhs_yb2 = np.zeros(n_cw*n_sw)
+    fhs_xb2 = np.zeros(n_cw*n_sw)
+    fhs_zb2 = np.zeros(n_cw*n_sw)
+    fhs_xah = np.zeros(n_cw*n_sw)
+    fhs_yah = np.zeros(n_cw*n_sw)
+    fhs_zah = np.zeros(n_cw*n_sw)
+    fhs_xbh = np.zeros(n_cw*n_sw)
+    fhs_ybh = np.zeros(n_cw*n_sw)
+    fhs_zbh = np.zeros(n_cw*n_sw)
+    fhs_xch = np.zeros(n_cw*n_sw)
+    fhs_ych = np.zeros(n_cw*n_sw)
+    fhs_zch = np.zeros(n_cw*n_sw)
+    fhs_xc  = np.zeros(n_cw*n_sw)
+    fhs_yc  = np.zeros(n_cw*n_sw)
+    fhs_zc  = np.zeros(n_cw*n_sw)
+    fhs_xac = np.zeros(n_cw*n_sw)
+    fhs_yac = np.zeros(n_cw*n_sw)
+    fhs_zac = np.zeros(n_cw*n_sw)
+    fhs_xbc = np.zeros(n_cw*n_sw)
+    fhs_ybc = np.zeros(n_cw*n_sw)
+    fhs_zbc = np.zeros(n_cw*n_sw)
+    fhs_x   = np.zeros((n_cw+1)*(n_sw+1))
+    fhs_y   = np.zeros((n_cw+1)*(n_sw+1))
+    fhs_z   = np.zeros((n_cw+1)*(n_sw+1))      
+
+    fvs_xc    = np.zeros(n_cw*n_sw)
+    fvs_zc    = np.zeros(n_cw*n_sw)
+    fvs_yc    = np.zeros(n_cw*n_sw)   
+    fvs_x     = np.zeros((n_cw+1)*(n_sw+1))
+    fvs_y     = np.zeros((n_cw+1)*(n_sw+1))
+    fvs_z     = np.zeros((n_cw+1)*(n_sw+1))   
+    fus_v_cs  = np.zeros(n_sw)     
+    
+    # arrays to hold strip discretization values
+    leading_edge_indices    = np.array([],dtype=bool)    
+    trailing_edge_indices   = np.array([],dtype=bool)    
+    panels_per_strip        = np.array([],dtype=np.int16)
+    chordwise_panel_number  = np.array([],dtype=np.int16)
+    chord_lengths           = np.array([])               
+    tangent_incidence_angle = np.array([])               
+
+    # geometry values
+    semispan_h = fus.width * 0.5  
+    semispan_v = fus.heights.maximum * 0.5
+    origin     = fus.origin[0]
+
+    # Compute the curvature of the nose/tail given fineness ratio. Curvature is derived from general quadratic equation
+    # This method relates the fineness ratio to the quadratic curve formula via a spline fit interpolation
+    vec1               = [2 , 1.5, 1.2 , 1]
+    vec2               = [1  ,1.57 , 3.2,  8]
+    x                  = np.linspace(0,1,4)
+    fus_nose_curvature =  np.interp(np.interp(fus.fineness.nose,vec2,x), x , vec1)
+    fus_tail_curvature =  np.interp(np.interp(fus.fineness.tail,vec2,x), x , vec1) 
+
+    # Horizontal Sections of fuselage
+    fhs        = Data()        
+    fhs.origin = np.zeros((n_sw+1,3))        
+    fhs.chord  = np.zeros((n_sw+1))         
+    fhs.sweep  = np.zeros((n_sw+1))     
+                 
+    fvs        = Data() 
+    fvs.origin = np.zeros((n_sw+1,3))
+    fvs.chord  = np.zeros((n_sw+1)) 
+    fvs.sweep  = np.zeros((n_sw+1)) 
+
+    si         = np.arange(1,((n_sw*2)+2))
+    spacing    = np.cos((2*si - 1)/(2*len(si))*np.pi)     
+    h_array    = semispan_h*spacing[0:int((len(si)+1)/2)][::-1]  
+    v_array    = semispan_v*spacing[0:int((len(si)+1)/2)][::-1]  
+
+    for i in range(n_sw+1): 
```

Comment:
I would love to see an 
```
if segments in fuselage:
   then model that
else:
    your code 
```

that way we can do evtols/ a more refined 737.

Reply:
Yeah, tim actually talked to me about that a little. I actually only touched a couple lines in the fuselage function so that Emilio's arbitrary discretization would work. I noticed that you had some unused variables for generating a vertical fuselage in addition to the horizontal one we already make. Maybe it would also be good to make some kind of "VLM_fuselage_improvements" branch and PR?